### ISSUE TITLE
Application launches in incorrect workspace

### ISSUE THREAD

**Issue by ritiek** [linuxmint/Cinnamon#6885]
```
 * Cinnamon 3.4.6
 * Distribution - Linux Mint 18.2 Sonya
 * Graphics hardware - VGA compatible controller: Intel Corporation 3rd Gen Core processor Graphics Controller (rev 09) (prog-if 00 [VGA controller])
 * 64-bit
 ```

**Issue**

Opening a GUI application and then changing the workspace before the application actually shows up makes the application open in the new workspace rather than in the old workspace.

**Steps to reproduce**

1. Open any GUI application.
2. Change the workspace <kbd>ctrl</kbd>+<kbd>alt</kbd>+<kbd>→</kbd> before the application actually shows up.
3. The application shows up in the new workspace.

**Expected behaviour**

The application should open in the old workspace instead of new workspace.

**Other information**

**Comment by NikoKrause** [comment]
I think this depends on the application:
LibreOffice and Pix open in old workspace
Firefox opens on new workspace
Gimp switches you automatically back to the old workspace

**Comment by ritiek** [comment]
You're correct. I only tested with a few apps, they all opened in new workspace but I can confirm your results after some more testing.

**Comment by ritiek** [comment]
Is it possible to modify this behavior for all apps? I want them to open in old workspace by default.

**Comment by icarter09** [comment]
@ritiek are you still experiencing this issue?

**Comment by ritiek** [comment]
@icarter09 Yes! I was able to reproduce the issue right now on:
```
* Cinnamon version - Cinnamon 4.2.4
  - Not using daily builds PPA (https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds)
* Distribution - Linux Mint 19.2 Tina
* Graphics hardware -  VGA compatible controller: Intel Corporation Atom Processor Z36xxx/Z37xxx Series Graphics & Display (rev 0f)
* 64-bit
```
